File: mobile/android/core/src/main/kotlin/com/portalia/core/model/ResidentModels.kt
package com.portalia.core.model

import kotlinx.serialization.Serializable

@Serializable
data class ResidentLoginResponse(val accessToken: String, val refreshToken: String, val mustChangePassword: Boolean)

@Serializable
data class ResidentRefreshResponse(val accessToken: String, val refreshToken: String, val mustChangePassword: Boolean)

/** POST /resident-auth/change-password devolve só isto — sem mustChangePassword. */
@Serializable
data class ResidentChangePasswordResponse(val accessToken: String, val refreshToken: String)

@Serializable
data class ResidentUnitSummary(val id: String, val identifier: String, val block: String? = null)

@Serializable
data class ResidentCondominiumSummary(val id: String, val name: String)

@Serializable
data class ResidentMeResponse(
    val id: String,
    val name: String,
    val email: String? = null,
    val phone: String? = null,
    val unit: ResidentUnitSummary,
    val condominium: ResidentCondominiumSummary,
)

@Serializable
data class ResidentDashboardResponse(
    val pendingDeliveries: Int,
    val visitorsThisMonth: Int,
    val avgVisitsPerWeek: Double? = null,
    val avgVisitDurationMinutes: Int? = null,
)

@Serializable
data class ResidentTelephonyCredentials(
    val configured: Boolean,
    val sipUsername: String? = null,
    val sipPassword: String? = null,
    val domain: String? = null,
)

File: mobile/ios/PortaliaCore/Sources/PortaliaCore/Models/ResidentModels.swift
import Foundation

public struct ResidentLoginResponse: Decodable {
    public let accessToken: String
    public let refreshToken: String
    public let mustChangePassword: Bool
}

/// POST /resident-auth/change-password devolve só isto — sem `mustChangePassword`.
public struct ResidentChangePasswordResponse: Decodable {
    public let accessToken: String
    public let refreshToken: String
}

public struct ResidentRefreshResponse: Decodable {
    public let accessToken: String
    public let refreshToken: String
    public let mustChangePassword: Bool
}

public struct ResidentUnitSummary: Decodable {
    public let id: String
    public let identifier: String
    public let block: String?
}

public struct ResidentCondominiumSummary: Decodable {
    public let id: String
    public let name: String
}

public struct ResidentMeResponse: Decodable {
    public let id: String
    public let name: String
    public let email: String?
    public let phone: String?
    public let unit: ResidentUnitSummary
    public let condominium: ResidentCondominiumSummary
}

public struct ResidentDashboardResponse: Decodable {
    public let pendingDeliveries: Int
    public let visitorsThisMonth: Int
    public let avgVisitsPerWeek: Double?
    public let avgVisitDurationMinutes: Int?
}

public struct ResidentTelephonyCredentials: Decodable {
    public let configured: Bool
    public let sipUsername: String?
    public let sipPassword: String?
    public let domain: String?
}
